apps/src/sites/studio/pages/components/ScriptEditor.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import FlexGroup from './FlexGroup';
+
+const styles = {
+  input: {
+    width: '100%',
+    boxSizing: 'border-box',
+    padding: '4px 6px',
+    color: '#555',
+    border: '1px solid #ccc',
+    borderRadius: 4
+  },
+  checkbox: {
+    margin: '0 0 0 7px'
+  }
+};
+
+/**
+ * Component for editing course scripts.
+ */
+const ScriptEditor = React.createClass({
+  propTypes: {
+    scriptData: React.PropTypes.object.isRequired,
```

Comment:
I find that when passing in objects of data like this, it can be hard to follow what exactly is expected.

I wonder if it would be clearer to pass in each of the individual fields as individual props. (i.e. instead of scriptData, we have hidden, loginRequired, hideable_stages, etc). At the very least, it would probably be worth (a) describing the shape of this object and/or (b) destructing it in the beginning of your render, i.e. `const { hidden, loginRequired, hideable_stages } = this.props.stageData;`